app/screens/settings/notification_settings_mobile/index.js

```diff
@@ -1,19 +1,36 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {bindActionCreators} from 'redux';
 import {connect} from 'react-redux';
 
 import {getConfig} from 'mattermost-redux/selectors/entities/general';
-
 import {getTheme} from 'mattermost-redux/selectors/entities/preferences';
+import {updateMe} from 'mattermost-redux/actions/users';
 
 import NotificationSettingsMobile from './notification_settings_mobile';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/users';
 
 function mapStateToProps(state) {
+    const config = getConfig(state);
+    const theme = getTheme(state);
+    const updateMeRequest = state.requests.users.updateMe;
+    const currentUser = getCurrentUser(state) || {};
```

Comment:
Instead of using `|| {}` to provide a default value for this, can you instead use `defaultProps` to specify it? The current user should be defined by this point, but using `|| {}` like this could cause problems elsewhere because returning an object literal from `mapStateToProps` makes a new object each time. That can cause the component to re-render whenever anything in the store changes.

Reply:
Sure! But, in this case, wouldn't be better to use only `getCurrentUser(state)` and remove the `|| {}`? Or we go with the `defaultProps` solution?

(Just to let you know, there are few other places doing the same. I believe you guys would like to fix that):

![Screenshot 2019-04-24 19 23 23](https://user-images.githubusercontent.com/2411367/56697934-92905100-66c6-11e9-8d11-7fa89c8a32f4.png)

Reply:
I'm not sure if I got the solution with the `defaultProps`. You mean creating something like the code below?
```javascript
const defaultProps = {
    currentUser: {}
}

const currentUser = getCurrentUser(state) || defaultProps.currentUser;
```

Reply:
@hmhealey what do you think? Is it the way to go?

Reply:
Sorry, I should clarify. You can define a static `defaultProps` field on the component to set a default value for anything that is not set by `mapStateToProps` or whoever uses the component. It looks like this:
```javascript
class MyComponent extends React.Component {
    static defaultProps = {
        value: 1234,
    };
}

function mapStateToProps() {
    let value; // value is undefined
    return {
        value,
    };
}

export default connect(mapStateToProps)(MyComponent);
```

And I know that we do that in other places, but we haven't gotten around to changing them. Since the user has to be logged in by the time that most of these components are used, I haven't been in a real rush to remove them.

Reply:
Got it. Thanks for clarifying. Just updated the code.

Reply:
Thanks! Looks good now.

Reply:
👍